File: app/src/main/java/com/weiqi/ui/theme/Theme.kt
package com.weiqi.ui.theme

import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Shapes
import androidx.compose.material3.Typography
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.em
import androidx.compose.ui.unit.sp

/** Zen Go palette — "Digital Ink on Physical Wood." */
object Zen {
    val surface = Color(0xFFFFF8F3)
    val surfaceDim = Color(0xFFE7D8C6)
    val surfaceContainerLow = Color(0xFFFFF2E3)
    val surfaceContainer = Color(0xFFFBECDA)
    val surfaceContainerHigh = Color(0xFFF5E6D4)
    val surfaceContainerHighest = Color(0xFFEFE0CF)
    val onSurface = Color(0xFF221A10)
    val onSurfaceVariant = Color(0xFF504442)
    val outline = Color(0xFF827471)
    val outlineVariant = Color(0xFFD4C3BF)

    val primary = Color(0xFF361F1A)         // Walnut / Ink Black
    val onPrimary = Color(0xFFFFFFFF)
    val primaryContainer = Color(0xFF4E342E)
    val onPrimaryContainer = Color(0xFFC19C94)

    val secondary = Color(0xFF5F5E5F)       // Silt
    val secondaryContainer = Color(0xFFE2DFE0)
    val onSecondaryContainer = Color(0xFF636263)

    val tertiary = Color(0xFF242523)
    val tertiaryContainer = Color(0xFF3A3B39)
    val error = Color(0xFFBA1A1A)

    // Custom: board / stones.
    val kayaWood = Color(0xFFE8C99B)        // raised board
    val kayaWoodEdge = Color(0xFFCFA976)
    val tableShadow = Color(0x1F000000)
    val gridInk = Color(0xFF221A10)         // 20% opacity in renderer
    val blackStoneTop = Color(0xFF2B2B2B)
    val blackStoneBottom = Color(0xFF101010)
    val whiteStoneTop = Color(0xFFFFFCF6)
    val whiteStoneBottom = Color(0xFFE8DFD0)
}

private val ZenColors = lightColorScheme(
    primary = Zen.primary,
    onPrimary = Zen.onPrimary,
    primaryContainer = Zen.primaryContainer,
    onPrimaryContainer = Zen.onPrimaryContainer,
    secondary = Zen.secondary,
    onSecondary = Color.White,
    secondaryContainer = Zen.secondaryContainer,
    onSecondaryContainer = Zen.onSecondaryContainer,
    tertiary = Zen.tertiary,
    error = Zen.error,
    background = Zen.surface,
    onBackground = Zen.onSurface,
    surface = Zen.surface,
    onSurface = Zen.onSurface,
    surfaceVariant = Zen.surfaceContainerHighest,
    onSurfaceVariant = Zen.onSurfaceVariant,
    outline = Zen.outline,
    outlineVariant = Zen.outlineVariant,
    surfaceContainerLowest = Color.White,
    surfaceContainerLow = Zen.surfaceContainerLow,
    surfaceContainer = Zen.surfaceContainer,
    surfaceContainerHigh = Zen.surfaceContainerHigh,
    surfaceContainerHighest = Zen.surfaceContainerHighest,
)

private val Serif = FontFamily.Serif      // Falls back gracefully without bundled Noto Serif.
private val Sans = FontFamily.SansSerif   // Stand-in for Manrope.

private val ZenTypography = Typography(
    displayLarge = TextStyle(
        fontFamily = Serif, fontWeight = FontWeight.Bold,
        fontSize = 42.sp, lineHeight = 52.sp, letterSpacing = (-0.02).em
    ),
    headlineMedium = TextStyle(
        fontFamily = Serif, fontWeight = FontWeight.SemiBold,
        fontSize = 24.sp, lineHeight = 32.sp
    ),
    headlineSmall = TextStyle(
        fontFamily = Serif, fontWeight = FontWeight.SemiBold,
        fontSize = 20.sp, lineHeight = 28.sp
    ),
    titleLarge = TextStyle(
        fontFamily = Serif, fontWeight = FontWeight.SemiBold,
        fontSize = 22.sp, lineHeight = 28.sp
    ),
    bodyLarge = TextStyle(
        fontFamily = Sans, fontWeight = FontWeight.Normal,
        fontSize = 18.sp, lineHeight = 28.sp
    ),
    bodyMedium = TextStyle(
        fontFamily = Sans, fontWeight = FontWeight.Normal,
        fontSize = 16.sp, lineHeight = 24.sp
    ),
    labelLarge = TextStyle(
        fontFamily = Sans, fontWeight = FontWeight.SemiBold,
        fontSize = 14.sp, lineHeight = 20.sp, letterSpacing = 0.05.em
    ),
    labelMedium = TextStyle(
        fontFamily = Sans, fontWeight = FontWeight.SemiBold,
        fontSize = 14.sp, lineHeight = 20.sp, letterSpacing = 0.05.em
    ),
    labelSmall = TextStyle(
        fontFamily = Sans, fontWeight = FontWeight.Medium,
        fontSize = 12.sp, lineHeight = 16.sp, letterSpacing = 0.08.em
    ),
)

// Softer rounding throughout — gives a warm, hand-carved feel without becoming bubbly.
private val ZenShapes = Shapes(
    extraSmall = RoundedCornerShape(8.dp),
    small = RoundedCornerShape(12.dp),
    medium = RoundedCornerShape(14.dp),
    large = RoundedCornerShape(18.dp),
    extraLarge = RoundedCornerShape(24.dp),
)

@Composable
fun WeiqiTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = ZenColors,
        typography = ZenTypography,
        shapes = ZenShapes,
        content = content
    )
}

File: lib/src/ui/theme.dart
import 'package:flutter/material.dart';

/// "Digital Ink on Physical Wood" — Zen palette and typography.
class Zen {
  static const double cardRadius = 18;
  static const double controlRadius = 14;
  static const double dialogRadius = 20;

  static const surface = Color(0xFFFFF8F3);
  static const surfaceDim = Color(0xFFE7D8C6);
  static const surfaceContainerLow = Color(0xFFFFF2E3);
  static const surfaceContainer = Color(0xFFFBECDA);
  static const surfaceContainerHigh = Color(0xFFF5E6D4);
  static const surfaceContainerHighest = Color(0xFFEFE0CF);
  static const onSurface = Color(0xFF221A10);
  static const onSurfaceVariant = Color(0xFF504442);
  static const outline = Color(0xFF827471);
  static const outlineVariant = Color(0xFFD4C3BF);

  static const primary = Color(0xFF361F1A);
  static const onPrimary = Color(0xFFFFFFFF);
  static const primaryContainer = Color(0xFF4E342E);
  static const onPrimaryContainer = Color(0xFFC19C94);

  static const secondary = Color(0xFF5F5E5F);
  static const secondaryContainer = Color(0xFFE2DFE0);
  static const onSecondaryContainer = Color(0xFF636263);

  static const tertiary = Color(0xFF242523);
  static const tertiaryContainer = Color(0xFF3A3B39);
  static const error = Color(0xFFBA1A1A);

  // Custom: board / stones.
  static const kayaWood = Color(0xFFE8C99B);
  static const kayaWoodEdge = Color(0xFFCFA976);
  static const tableShadow = Color(0x1F000000);
  static const gridInk = Color(0xFF221A10);
  static const blackStoneTop = Color(0xFF2B2B2B);
  static const blackStoneBottom = Color(0xFF101010);
  static const whiteStoneTop = Color(0xFFFFFCF6);
  static const whiteStoneBottom = Color(0xFFE8DFD0);
}

ThemeData buildWeiqiTheme() {
  const scheme = ColorScheme(
    brightness: Brightness.light,
    primary: Zen.primary,
    onPrimary: Zen.onPrimary,
    primaryContainer: Zen.primaryContainer,
    onPrimaryContainer: Zen.onPrimaryContainer,
    secondary: Zen.secondary,
    onSecondary: Colors.white,
    secondaryContainer: Zen.secondaryContainer,
    onSecondaryContainer: Zen.onSecondaryContainer,
    tertiary: Zen.tertiary,
    onTertiary: Colors.white,
    error: Zen.error,
    onError: Colors.white,
    surface: Zen.surface,
    onSurface: Zen.onSurface,
    surfaceContainerLowest: Colors.white,
    surfaceContainerLow: Zen.surfaceContainerLow,
    surfaceContainer: Zen.surfaceContainer,
    surfaceContainerHigh: Zen.surfaceContainerHigh,
    surfaceContainerHighest: Zen.surfaceContainerHighest,
    surfaceDim: Zen.surfaceDim,
    outline: Zen.outline,
    outlineVariant: Zen.outlineVariant,
  );

  const serif = 'serif';
  const sans = 'sans-serif';

  final textTheme = TextTheme(
    displayLarge: TextStyle(
      fontFamily: serif,
      fontWeight: FontWeight.bold,
      fontSize: 42,
      height: 52 / 42,
      letterSpacing: 0,
      color: scheme.onSurface,
    ),
    headlineMedium: TextStyle(
      fontFamily: serif,
      fontWeight: FontWeight.w600,
      fontSize: 24,
      height: 32 / 24,
      color: scheme.onSurface,
    ),
    headlineSmall: TextStyle(
      fontFamily: serif,
      fontWeight: FontWeight.w600,
      fontSize: 20,
      height: 28 / 20,
      color: scheme.onSurface,
    ),
    titleLarge: TextStyle(
      fontFamily: serif,
      fontWeight: FontWeight.w600,
      fontSize: 22,
      height: 28 / 22,
      color: scheme.onSurface,
    ),
    bodyLarge: TextStyle(
      fontFamily: sans,
      fontSize: 18,
      height: 28 / 18,
      color: scheme.onSurface,
    ),
    bodyMedium: TextStyle(
      fontFamily: sans,
      fontSize: 16,
      height: 24 / 16,
      color: scheme.onSurface,
    ),
    labelLarge: TextStyle(
      fontFamily: sans,
      fontWeight: FontWeight.w600,
      fontSize: 14,
      height: 20 / 14,
      letterSpacing: 0,
      color: scheme.onSurface,
    ),
    labelMedium: TextStyle(
      fontFamily: sans,
      fontWeight: FontWeight.w600,
      fontSize: 14,
      height: 20 / 14,
      letterSpacing: 0,
      color: scheme.onSurface,
    ),
    labelSmall: TextStyle(
      fontFamily: sans,
      fontWeight: FontWeight.w500,
      fontSize: 12,
      height: 16 / 12,
      letterSpacing: 0,
      color: scheme.onSurface,
    ),
  );

  return ThemeData(
    useMaterial3: true,
    colorScheme: scheme,
    scaffoldBackgroundColor: Zen.surface,
    canvasColor: Zen.surface,
    textTheme: textTheme,
    appBarTheme: const AppBarTheme(
      backgroundColor: Zen.surface,
      foregroundColor: Zen.onSurface,
      centerTitle: true,
      elevation: 0,
      scrolledUnderElevation: 0,
    ),
    navigationBarTheme: const NavigationBarThemeData(
      backgroundColor: Zen.surfaceContainer,
      indicatorColor: Zen.surfaceContainerHigh,
      surfaceTintColor: Colors.transparent,
      elevation: 0,
    ),
    filledButtonTheme: FilledButtonThemeData(
      style: FilledButton.styleFrom(
        backgroundColor: scheme.primary,
        foregroundColor: scheme.onPrimary,
        minimumSize: const Size(0, 54),
        padding: const EdgeInsets.symmetric(horizontal: 24),
        textStyle: textTheme.labelLarge?.copyWith(
          fontWeight: FontWeight.w700,
          letterSpacing: 0.3,
        ),
        shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(Zen.controlRadius)),
      ),
    ),
    outlinedButtonTheme: OutlinedButtonThemeData(
      style: OutlinedButton.styleFrom(
        foregroundColor: scheme.onSurface,
        side: BorderSide(color: scheme.outline.withValues(alpha: 0.4)),
        shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(Zen.controlRadius)),
      ),
    ),
    chipTheme: ChipThemeData(
      shape: RoundedRectangleBorder(
          borderRadius: BorderRadius.circular(Zen.controlRadius)),
      backgroundColor: scheme.surfaceContainerHigh,
      selectedColor: scheme.primary,
      labelStyle: textTheme.labelLarge,
    ),
    sliderTheme: SliderThemeData(
      activeTrackColor: scheme.primary,
      inactiveTrackColor: scheme.surfaceContainerHigh,
      thumbColor: scheme.primary,
    ),
    switchTheme: SwitchThemeData(
      thumbColor: WidgetStateProperty.resolveWith((s) =>
          s.contains(WidgetState.selected) ? scheme.primary : Colors.white),
      trackColor: WidgetStateProperty.resolveWith((s) =>
          s.contains(WidgetState.selected)
              ? scheme.primary.withValues(alpha: 0.5)
              : scheme.surfaceContainerHigh),
    ),
    cardTheme: CardThemeData(
      color: scheme.surfaceContainer,
      elevation: 0,
      shape: RoundedRectangleBorder(
          borderRadius: BorderRadius.circular(Zen.cardRadius)),
    ),
    dialogTheme: DialogThemeData(
      backgroundColor: scheme.surfaceContainerLow,
      shape: RoundedRectangleBorder(
          borderRadius: BorderRadius.circular(Zen.dialogRadius)),
    ),
  );
}
